lightning/src/ln/channel.rs

```diff
@@ -495,6 +495,7 @@ pub(super) struct Channel<Signer: Sign> {
 	#[cfg(not(any(test, feature = "_test_utils")))]
 	config: ChannelConfig,
 
+	announced: bool,
```

Comment:
Why is this third instance of an `announced` bool (besides the new `ChannelHandshakeConfig::announced_channel` and the now-deprecated `ChannelConfig::announced_channel`) necessary? Since it seems to be instantiated with `own_channel_config.announced_channel`, couldn't we just use this or move it entirely to `Channel`?

Reply:
`ChannelHandshakeConfig` doesn't need to be stored anywhere (and currently isn't) as it's mostly ephemeral, but the `announced` bit should definitely continue to be stored, so I opted to have it within the `Channel` struct.